components/automate-cli/pkg/diagnostics/integration/iam_v2_policies.go

```diff
@@ -0,0 +1,218 @@
+package integration
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+
+	"github.com/chef/automate/components/automate-cli/pkg/diagnostics"
+	"github.com/chef/automate/components/automate-cli/pkg/diagnostics/lbrequest"
+)
+
+const createV2PolicyTemplateStr = `
+{
+	"id": "{{ .ID }}",
+	"name": "{{ .ID }} test policy",
+	"members": [
+		"token:{{ .TokenID }}", 
+		"user:local:{{ .UserID}}", 
+		"team:local:{{ .TeamID }}"
+	],
+	"statements": [
+		{
+			"effect": "ALLOW",
+			"role": "{{ .RoleID }}",
+			"projects": ["{{ .ProjectID }}"]
+		}
+	]
+}
+`
+
+// V2PolicyInfo represents the nested response from the v2 Policy API.
+type V2PolicyInfo struct {
+	Policy PolicyInfo
+}
+
+// PolicyParameters represents policy inputs
+type PolicyParameters struct {
+	TokenID   string
+	TeamID    string
+	UserID    string
+	RoleID    string
+	ProjectID string
+}
+
+type iamV2PoliciesSave struct {
+	TokenID    string `json:"token_id"`
+	TokenValue string `json:"token_value"`
+	PolicyID   string `json:"policy_id"`
+	TeamID     string `json:"team_id"`
+	UserID     string `json:"user_id"`
+	RoleID     string `json:"role_id"`
+	ProjectID  string `json:"project_id"`
+}
+
+// CreateV2Policy creates a policy using the given policy fields
+func CreateV2Policy(tstCtx diagnostics.TestContext, pol PolicyParameters) (*PolicyInfo, error) {
+	policyInfo := PolicyInfo{}
+	v2PolicyInfo := V2PolicyInfo{Policy: policyInfo}
+	err := MustJSONDecodeSuccess(
+		tstCtx.DoLBRequest(
+			"/apis/iam/v2/policies",
+			lbrequest.WithMethod("POST"),
+			lbrequest.WithJSONStringTemplateBody(createV2PolicyTemplateStr, struct {
+				ID        string
+				TokenID   string
+				UserID    string
+				TeamID    string
+				RoleID    string
+				ProjectID string
+			}{
+				ID:        TimestampName(),
+				TokenID:   pol.TokenID,
+				UserID:    pol.UserID,
+				TeamID:    pol.TeamID,
+				RoleID:    pol.RoleID,
+				ProjectID: pol.ProjectID,
+			}),
+		)).WithValue(&v2PolicyInfo)
+	policyInfo = v2PolicyInfo.Policy
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not create policy")
+	}
+
+	return &policyInfo, nil
+}
+
+// CreateIAMV2PoliciesDiagnostic create the diagnostic struct for v2 policies.
+func CreateIAMV2PoliciesDiagnostic() diagnostics.Diagnostic {
```

Comment:
this is our most holistic diagnostic, since it creates each possible policy component (members, role, project) and uses them to create the policy.